pkg/cmd/server/server.go

```diff
@@ -584,6 +604,27 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 		)
 		cmd.CheckError(err)
 
+		// Make empty listers that will only be populated if CSI is properly enabled.
+		var vsLister snapshotv1beta1listers.VolumeSnapshotLister
+		var vscLister snapshotv1beta1listers.VolumeSnapshotContentLister
+
+		// We don't care about the returned types, just whether or not there's a 404 error.
+		_, err = s.discoveryClient.ServerResourcesForGroupVersion(snapshotv1beta1api.SchemeGroupVersion.String())
+		switch {
+		case apierrors.IsNotFound(err) && !features.IsEnabled("EnableCSI"):
```

Comment:
Should we wrap this whole block (L611-626) in a `if features.IsEnabled("EnableCSI")`, so we don't even execute the discoveryClient call if it's not enabled? Then we can remove this first case and remove the check on the feature flag from all the others. Simpler and also avoids the code path for when the feature flag is not enabled.